sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py

```diff
@@ -43,20 +44,22 @@ class LogsQueryClient(object): # pylint: disable=client-accepts-api-version-keyw
 
     :param credential: The credential to authenticate the client.
     :type credential: ~azure.core.credentials.TokenCredential
-    :keyword endpoint: The endpoint to connect to. Defaults to 'https://api.loganalytics.io'.
+    :keyword endpoint: The endpoint to connect to. Defaults to 'https://api.loganalytics.io/v1'.
     :paramtype endpoint: Optional[str]
     """
 
     def __init__(self, credential: TokenCredential, **kwargs: Any) -> None:
-        endpoint = kwargs.pop("endpoint", "https://api.loganalytics.io")
+        endpoint = kwargs.pop("endpoint", "https://api.loganalytics.io/v1")
         if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
             endpoint = "https://" + endpoint
+        parsed_endpoint = urlparse(endpoint)
```

Comment:
For a user defined endpoint too, we should be appending the version v1. JS customer has reported an issue on this - https://github.com/Azure/azure-sdk-for-js/issues/22408

Reply:
I think the guidance for that issue should be that the user pass in `"https://api.loganalytics.io/v1"` as the endpoint, but then the client should strip the v1 and provide that as the audience for the credential scope (i.e., `https://api.loganalytics.io/.default`). Here a user can also explicitly pass in an audience in the off chance that the hostname differs from the endpoint hostname.